extensions/arc/src/ui/dashboards/postgres/postgresResourceHealthPage.ts

```diff
@@ -0,0 +1,194 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as loc from '../../../localizedConstants';
+import { IconPathHelper, cssStyles } from '../../../constants';
+import { DashboardPage } from '../../components/dashboardPage';
+import { PostgresModel } from '../../../models/postgresModel';
+import { fromNow } from '../../../common/date';
+
+export class PostgresResourceHealthPage extends DashboardPage {
+	private disposables: vscode.Disposable[] = [];
+	private interval: NodeJS.Timeout;
+	private podsUpdated?: azdata.TextComponent;
+	private podsTable?: azdata.DeclarativeTableComponent;
+	private conditionsTable?: azdata.DeclarativeTableComponent;
+
+	constructor(protected modelView: azdata.ModelView, private _postgresModel: PostgresModel) {
+		super(modelView);
+
+		modelView.onClosed(() => {
+			try { clearInterval(this.interval); }
```

Comment:
It did.  Even with the event listener fixes, this interval would keep triggering after I closed the page.

Reply:
I can't be certain that the pages are garbage collected even with these fixes.  I just know that the 2 places where it's obvious they're still around (the event listeners and interval) are no longer firing.

Reply:
Oh yeah that makes sense - didn't think of it that way. Here's a useful article on it - https://blog.x5ff.xyz/blog/typescript-interval-iot/

Since the callback has a reference to this then it'd be kept around forever and thus the page would never get GCd. 

They use EventEmitter in their example but I'd expect that clearing the interval would accomplish the same thing.